### PR TITLE
test(shell): add unit tests for shell_validation module

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

- Add **114 unit tests** for `gptme/tools/shell_validation.py`, which previously had **zero test coverage**
- Covers all public and internal functions: quote parsing, heredoc detection, pipe finding, redirection detection, allowlist/denylist checking
- All tests are pure unit tests with no external dependencies

## Test categories

| Category | Tests | Functions covered |
|----------|-------|-------------------|
| Quote parsing | 12 | `_find_quotes` |
| Heredoc detection | 7 | `_find_heredoc_regions` |
| Position checking | 6 | `_is_in_quoted_region` |
| Pipe detection | 8 | `_find_first_unquoted_pipe` |
| Redirection detection | 8 | `_has_file_redirection` |
| Allowlist | 14 | `is_allowlisted` |
| Denylist | 42 | `is_denylisted` |
| Integration | 5 | allowlist + denylist interaction |
| Edge cases | 12 | various boundary conditions |

## Test plan

- [x] All 114 tests pass locally
- [x] mypy clean
- [x] ruff formatted
- [ ] CI passes

### CODE REVIEW

File: tests/test_tools_shell_validation.py [hunk 1]
```diff
@@ -0,0 +1,620 @@
...
+    def test_git_add_dot(self):
+        denied, reason, _ = is_denylisted("git add .")
+        assert denied
+        assert reason is not None
+        assert "selective" in reason.lower() or "specific" in reason.lower()
+
+    def test_git_add_dot_not_dotfile(self):
+        # git add .gitignore should NOT be denied
+        denied, _, _ = is_denylisted("git add .gitignore")
+        assert not denied
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_tools_shell_validation.py:321]: Fixed in 70dcf66 — removed the duplicate `test_git_add_dotgitignore_not_denied`.

File: tests/test_tools_shell_validation.py [hunk 2]
```diff
@@ -0,0 +1,620 @@
...
+
+    def test_dangerous_find_flags(self):
+        assert not is_allowlisted("find . -exec rm {} \\;")
+        assert not is_allowlisted("find . -execdir cmd {} \\;")
+        assert not is_allowlisted("find . -delete")
+        assert not is_allowlisted("find . -ok rm {} \\;")
+
+    def test_safe_find(self):
+        assert is_allowlisted("find . -name '*.py'")
+        assert is_allowlisted("find . -type f")
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_tools_shell_validation.py:286]: Fixed in 70dcf66 — added `test_find_executable_false_positive` documenting that `find -executable` is incorrectly blocked by the `-exec` substring check.

**Comment by TimeToBuildBob** [comment]
Note: PR #1792 fixes the `find -executable` false positive that `test_find_executable_false_positive` documents. After #1792 merges, that test's assertion should be flipped from `assert not is_allowlisted(...)` to `assert is_allowlisted(...)`.
